bin/server/cmd/component_omni.go

```diff
@@ -48,19 +54,70 @@ func createMetadataDB(activeProfile *Profile, logger *zap.Logger) (*metadataDB,
 }
 
 func (m *metadataDB) connect() (*store.DB, error) {
-	if err := m.pgInstance.Start(m.profile.datastorePort, os.Stderr, os.Stderr); err != nil {
+	if useEmbedDB() {
+		if err := m.pgInstance.Start(m.profile.datastorePort, os.Stderr, os.Stderr); err != nil {
+			return nil, err
+		}
+		m.pgStarted = true
+
+		// Even when Postgres opens Unix domain socket only for connection, it still requires a port as ID to differentiate different Postgres instances.
+		connCfg := dbdriver.ConnectionConfig{
+			Username:    m.profile.pgUser,
+			Password:    "",
+			Host:        common.GetPostgresSocketDir(),
+			Port:        fmt.Sprintf("%d", m.profile.datastorePort),
+			StrictUseDb: false,
+		}
+		db := store.NewDB(m.l, connCfg, m.profile.demoDataDir, readonly, version, m.profile.mode)
+		return db, nil
+	}
+
+	u, err := url.Parse(pgURL)
+	if err != nil {
 		return nil, err
 	}
-	m.pgStarted = true
 
-	// Even when Postgres opens Unix domain socket only for connection, it still requires a port as ID to differentiate different Postgres instances.
+	m.l.Info("Establishing external PostgreSQL connection...", zap.String("pgURL", u.Redacted()))
+
+	if u.Scheme != "postgres" && u.Scheme != "postgresql" {
```

Comment:
According to the doc, only postgresql is supported

Reply:
my bad